pkg/controller/controller_test.go

```diff
@@ -1852,6 +1907,232 @@ func TestCatalogConversionClusterServicePlanBindable(t *testing.T) {
 
 }
 
+const testCatalogForClusterServiceClassAndPlanWithInvalidExternalIDCharacters = `{
+  "services": [
+    {
+      "name": "mysql",
+      "id": "MySQL-12.34.",
+      "plans": [{
+        "name": "normal-characters",
+        "id": "mysql-100mb"
+      },
+			{
+				"name": "contains-escape-chars",
+				"id": "abz-class"
+			},
+      {
+        "name": "invalid-internal-characters",
+        "id": "invalid/characters"
+      },
+			{
+				"name": "invalid-starting-ending-characters",
+				"id": "InvalidstarT"
+      },
+			{
+				"name": "starting-ending-periods",
+				"id": ".start."
+      },
+			{
+				"name": "period-next-to-invalid-character",
+				"id": "foo.Bar"
+      },
+			{
+				"name": "internal-period",
+				"id": "foo.bar"
+      },
+			{
+				"name": "too-long",
+				"id": "thisnameisreallyreallywaytoolonghowcouldanyonepossiblyreadthisplansname"
+      },
+			{
+				"name": "too-long-with-problem-chars",
+				"id": "thisnameisreallyreallywaytool-onghowcouldanyonepossiblyreadthisplansname"
+      },
+			{
+				"name": "too-long-with-problem-period",
+				"id": "thisnameisreallyreallywaytool.onghowcouldanyonepossiblyreadthisplansname"
+			}]
+    }
+]}`
+
+func TestCatalogConversionInvalidCharactersInExternalID(t *testing.T) {
```

Comment:
This is the meat of testing generateEscapedName() right?  For me it borders on incomprehensible for verifying escaped strings are as expected.  How about adding unit tests specific to only generateEscapedName()?